controllers/auth.js

```diff
@@ -0,0 +1,116 @@
+const userModelSchema = require("../model/user");
+const bcrypt = require("bcrypt")
+const jwt = require("jsonwebtoken");
+const { Email } = require("../utils/Email");
+//signup user
+const signup = async (req, res) => {
+  try {
+    const { firstName, lastName, email, password } = req.body;
+    const hashedPassword = await bcrypt.hash(password, 10);
+    const user = await userModelSchema.findOne({ email: email });
+    if (user) {
+      throw new Error("email already exist", 422);
+    }
+    const newUser = await userModelSchema.create({
+      firstName: firstName,
+      lastName: lastName,
+      email: email,
+      password: hashedPassword,
+    });
+    return res.status(200).json({
+      registerUser: newUser,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      error: "user already exist!",
+    });
+  }
+};
+
+// forgot password page action
+const forgotPassword = async (req, res, next) => {
+  try {
+    const { email } = req.body;
+    const user = await userModelSchema.findOne({ email: email });
+    // set the secret with jwt token + password
+    const secret = process.env.jwt_secret + user.password;
+    const payLoad = {
+      email: user.email,
+      id: user._id,
+    };
+    const token = jwt.sign(payLoad, secret, { expiresIn: '60m' });
+    const link = `http://localhost:8080/user/newpassword/${user._id}/${token}`;
+    // send email to set the new password
+    const emailClient = new Email();
+    emailClient.setBody(link);
+    emailClient.setSubject("verify your email");
```

Comment:
why the subject is verify you email?
are you sending email for email verification?

Reply:
> why the subject is verify you email? are you sending email for email verification?

no mam it's for token verification I have changed the subject and replaced it with token verification!.